packages/valory/connections/ledger/connection.py

```diff
@@ -162,37 +161,15 @@ async def receive(self, *args: Any, **kwargs: Any) -> Optional["Envelope"]:
         :param kwargs: the keyword arguments
         :return: the envelope received, or None.
         """
-        # if there are done tasks, return the result
-        if len(self.done_tasks) > 0:  # pragma: nocover
-            done_task = self.done_tasks.pop()
-            return self._handle_done_task(done_task)
-
-        if len(self.receiving_tasks) == 0:  # pragma: nocover
-            self.event_new_receiving_task.clear()
-            await self.event_new_receiving_task.wait()
-
-        # wait for completion of at least one receiving task
-        done, _ = await asyncio.wait(
-            self.receiving_tasks, return_when=asyncio.FIRST_COMPLETED
-        )
-
-        # pick one done task
-        done_task = done.pop()
+        return await self.response_envelopes.get()
```

Comment:
would it be better to use a timeout here?
 `await asyncio.wait_for(self.agent.outbox.get(), timeout=1)`

Reply:
The whole point is that it doesn't time out.